RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByMVAIsolationRun2.cc

```diff
@@ -221,74 +223,109 @@ double PFRecoTauDiscriminationByMVAIsolationRun2::discriminate(const PFTauRef& t
-      mvaInput_[11] = std::min((float)0.5, ptWeightedDrIsolation);
-      mvaInput_[12] = std::min((float)100., leadingTrackChi2);
-      mvaInput_[13] = std::min((float)1., eRatio);
-      mvaInput_[14]  = TMath::Sign((float)+1., (float)tauLifetimeInfo.dxy());
-      mvaInput_[15]  = std::sqrt(std::fabs(std::min((float)1., std::fabs((float)tauLifetimeInfo.dxy()))));
-      mvaInput_[16]  = std::min((float)10., std::fabs((float)tauLifetimeInfo.dxy_Sig()));
-      mvaInput_[17]  = TMath::Sign((float)+1., (float)tauLifetimeInfo.ip3d());
-      mvaInput_[18]  = std::sqrt(std::fabs(std::min((float)1., std::fabs((float)tauLifetimeInfo.ip3d()))));
-      mvaInput_[19]  = std::min((float)10., std::fabs((float)tauLifetimeInfo.ip3d_Sig()));
+      mvaInput_[7]  = std::min(30.f, nPhoton);
+      mvaInput_[8]  = std::min(0.5f, ptWeightedDetaStrip);
+      mvaInput_[9]  = std::min(0.5f, ptWeightedDphiStrip);
+      mvaInput_[10] = std::min(0.5f, ptWeightedDrSignal);
+      mvaInput_[11] = std::min(0.5f, ptWeightedDrIsolation);
+      mvaInput_[12] = std::min(100.f, leadingTrackChi2);
+      mvaInput_[13] = std::min(1.f, eRatio);
+      mvaInput_[14]  = std::copysign(+1.f, (float)tauLifetimeInfo.dxy());
+      mvaInput_[15]  = std::sqrt(std::fabs(std::min(1.f, std::fabs((float)tauLifetimeInfo.dxy()))));
+      mvaInput_[16]  = std::min(10.f, std::fabs((float)tauLifetimeInfo.dxy_Sig()));
+      mvaInput_[17]  = std::copysign(+1.f, (float)tauLifetimeInfo.ip3d());
+      mvaInput_[18]  = std::sqrt(std::fabs(std::min(1.f, std::fabs((float)tauLifetimeInfo.ip3d()))));
+      mvaInput_[19]  = std::min(10.f, std::fabs((float)tauLifetimeInfo.ip3d_Sig()));
       mvaInput_[20]  = ( tauLifetimeInfo.hasSecondaryVertex() ) ? 1. : 0.;
       mvaInput_[21] = std::sqrt(decayDistMag);
-      mvaInput_[22] = std::min((float)10., (float)tauLifetimeInfo.flightLengthSig());
+      mvaInput_[22] = std::min(10.f, (float)tauLifetimeInfo.flightLengthSig());
+    } else if ( mvaOpt_ == kDBoldDMwLTwGJ || mvaOpt_ == kDBnewDMwLTwGJ ) {
+      mvaInput_[0]  = std::log(std::max(1.f, (float)tau->pt()));
+      mvaInput_[1]  = std::fabs((float)tau->eta());
+      mvaInput_[2]  = std::log(std::max(1.e-2f, chargedIsoPtSum));
+      mvaInput_[3]  = std::log(std::max(1.e-2f, neutralIsoPtSum));
+      mvaInput_[4]  = std::log(std::max(1.e-2f, puCorrPtSum));
+      mvaInput_[5]  = std::log(std::max(1.e-2f, photonPtSumOutsideSignalCone));
+      mvaInput_[6]  = tauDecayMode;
+      mvaInput_[7]  = std::min(30.f, nPhoton);
+      mvaInput_[8]  = std::min(0.5f, ptWeightedDetaStrip);
+      mvaInput_[9]  = std::min(0.5f, ptWeightedDphiStrip);
+      mvaInput_[10] = std::min(0.5f, ptWeightedDrSignal);
+      mvaInput_[11] = std::min(0.5f, ptWeightedDrIsolation);
+      mvaInput_[12] = std::min(1.f, eRatio);
+      mvaInput_[13]  = std::copysign(+1.f, (float)tauLifetimeInfo.dxy());
+      mvaInput_[14]  = std::sqrt(std::fabs(std::min(1.f, std::fabs((float)tauLifetimeInfo.dxy()))));
```

Comment:
std::abs could be used
also, the outer-most one is redundant and is not present in the PAT version of the code